New vehicle parts and vehicles for Aftershock

Ilysen opened the pull request:
#### Summary
```SUMMARY: Mods "Added the control station and hauling space to various vehicles in Aftershock, as well as the new tanker trucks and mounted tank parts."```

#### Purpose of change
After playtesting much larger cargo spaces, I found them much more enjoyable to use, *but* I think changing the existing balance of the cargo space wouldn't go over that well. Control stations are mainly for compactness.

#### Describe the solution
* Added the control station to APCs. It combines the functionality of controls, a clock, remote controls, and a camera control station.
* Replaced Truck Trailers' cargo spaces with hauling spaces that can hold up to 1500 L of items.

The control station can't be crafted; you'll have to pull it off an existing APC. The hauling space *can* be crafted, using three heavy duty frames, four sheet metal, a welding torch, and some tools.

* Added roof-mounted external tanks and mounted external tanks; one goes on the roof, the other goes on a frame. Both hold 200 L. Two new vehicles use the mounted external tanks, the gas tanker and the water tanker.

#### Describe alternatives you've considered
Like I mentioned in the purpose, I considered changing existing cargo spaces, but cargo spaces right now are used to represent flatbeds, truck trailers, and pickups, among other things. It was simpler to replace the truck trailer's when I used that for my math.

#### Additional context
Hauling space capacity is based off of dimensions from this site: http://yrc.com/trailer-dimensions/

The smallest trailer on that site (28 foot high cube trailer) has a cubic capacity of 2029 cft, which converts to 57454.88 liters. A truck trailer has 32 spaces for cargo, so if you divide that number by 32, you get 1795.465 L of space per tile, which I rounded down to 1500 L so we can at least pretend it's balanced.

mlangsdorf commented on data/mods/Aftershock/vehicles/afs_vehicle_overrides.json:
```diff
@@ -0,0 +1,343 @@
+    "name": "Truck Trailer",
+    "blueprint": [ [ "--OO------" ], [ "||OO||||||" ], [ "+|++|=|++|" ], [ "+|++|=|++|" ], [ "||OO||||||" ], [ "--OO------" ] ],
+    "parts": [
+      { "x": 0, "y": 0, "part": "hdframe_vertical_2" },
```
Please consolidate multiple parts on the same tile with `"parts": [ ]` like you did with the APCs above.

Ilysen replied:
This was actually just a copypaste from the normal vehicles, with instances of `cargo_space` replaced with `afs_hauling_space`. I'll look into changing it to be less icky.

mlangsdorf commented on data/mods/Aftershock/vehicles/afs_part_items.json:
```diff
@@ -0,0 +1,92 @@
+    "type": "GENERIC",
+    "id": "afs_control_station",
+    "name": "control station",
+    "description": "A large and complex set of various vehicle controls.  Useful for a vehicle.",
```
The description could use more detail, specifying the functions it includes.

mlangsdorf commented on data/mods/Aftershock/vehicles/afs_vehicle_parts.json:
```diff
@@ -161,6 +176,41 @@
   },
+  {
+    "type": "vehicle_part",
+    "id": "afs_control_station",
```
I'd be fine with seeing this in mainline and expanded to a couple of other military vehicles.

I'd also like a version of this without REMOTE_CONTROLS - call it an integrated control console - added to all the military vehicles in Tanks, if that's something you would be interested in.

Ilysen replied:
I can put it in mainline, but I'd also open it as a different PR so as not to muddy the waters. I've never actually played using Tanks, but I don't think it should be too much of a stretch to put it in.

Ilysen commented:
I ended up expanding it a little more. I added roof-mounted tanks and mounted tanks, and added them to two new vehicles, the gas tanker and the water tanker. Like truck trailers, they don't actually have any controls of their own, but lug around a huge amount of liquid.

kevingranade commented:
json-style please.
